mindsdb_sdk/knowledge_bases.py

```diff
@@ -0,0 +1,307 @@
+import copy
+import json
+from typing import Union, List
+
+import pandas as pd
+
+from mindsdb_sql.parser.dialects.mindsdb import CreateKnowledgeBase, DropKnowledgeBase
+from mindsdb_sql.parser.ast import Identifier, Star, Select, BinaryOperation, Constant, Insert
+
+from mindsdb_sdk.utils.sql import dict_to_binary_op
+from mindsdb_sdk.utils.objects_collection import CollectionBase
+
+from .models import Model
+from .tables import Table
+from .query import Query
+from .databases import Database
+
+
+class KnowledgeBase(Query):
+    def __init__(self, project, data):
+
+        self.project = project
+        self.name = data['name']
+
+        self.storage = None
+        if data['storage'] is not None:
+            # if name contents '.' there could be errors
```

Comment:
yes, it is related to this condition.
for storage we receive a string with database and table separated by dot: 'database_name.table_name'
if something from it has dot inside name then we won't have `len(parts) == 2`. In that case self.storage will be None
It should not affect to functionality because self.storage is not used in sdk, it is a just information for user, if they want to look at it.
and also data['storage'] could be None.